pkg/operator/controllers/subnets/subnet_nsg.go

```diff
@@ -30,6 +30,13 @@ func (r *reconcileManager) ensureSubnetNSG(ctx context.Context, s subnet.Subnet)
 		return err
 	}
 
+	if subnetObject.SubnetPropertiesFormat.NetworkSecurityGroup == nil {
+		r.log.Infof("Fixing NSG from nil to %s", correctNSGResourceID)
+		subnetObject.NetworkSecurityGroup = &mgmtnetwork.SecurityGroup{ID: &correctNSGResourceID}
+		err = r.subnets.CreateOrUpdate(ctx, s.ResourceID, subnetObject)
+		return err
+	}
+
 	if !strings.EqualFold(*subnetObject.NetworkSecurityGroup.ID, correctNSGResourceID) {
 		r.log.Infof("Fixing NSG from %s to %s", *subnetObject.NetworkSecurityGroup.ID, correctNSGResourceID)
```

Comment:
nit: why not combining the two into one block ? I don't like code duplication :)